pubsub/natsstreaming/natsstreaming_test.go

```diff
@@ -0,0 +1,518 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package natsstreaming
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/dapr/components-contrib/pubsub"
+)
+
+func TestParseNATSStreamingMetadata(t *testing.T) {
```

Comment:
most of sub-tests have the similar implementation. Please rewrite them as [Table driven tests](https://github.com/golang/go/wiki/TableDrivenTests).

Reply:
grouped sub-tests as table driven tests